### PR TITLE
feat: forgeup

### PR DESCRIPTION
**Pull request by transmissions11**

### CODE REVIEW

**Comment by transmissions11** [comment]
should be feature complete, needs more testing tho

**Comment by transmissions11** [comment]
should be able to do:

```
forgeup 

forgeup -v nightly 

forgeup -r transmissions11/foundry 

forgeup -b master
```

`-v` is ignored if the repo isn't `gakonst/foundry` and there is no branch specified

**Comment by onbjerg** [comment]
getting `sysctl: invalid option -- 'i'` on linux, still downloads nightly though so maybe not a big deal

everything else works as intended (-b, -r, switching repo, switching branch etc), lgtm

**Comment by transmissions11** [comment]
when do you see the sysctl error? just running plain `forgeup`?

**Comment by transmissions11** [comment]
oh its def from `> sysctl -in sysctl.proc_translated`

**Comment by onbjerg** [comment]
yes, I assume it is because macos has some -i flag on sysctl that is not standard. i think just `-n` should be fine?

**Comment by transmissions11** [comment]
yeah what even is the point of that i flag

**Comment by transmissions11** [comment]
Screen Shot 2022-01-16 at 8 03 37 PM

**Comment by onbjerg** [comment]
yea `man sysctl` gives me no `-i` flag here. assuming sysctl just does an `exit 1` if the oid doesn't exist then the `if` should fail and we should be good

**Comment by transmissions11** [comment]
yeah doesnt look like a big deal

Screen Shot 2022-01-16 at 8 04 46 PM

**Comment by transmissions11** [comment]
ok just pushed a version without -i, can you try it again and make sure it doesnt show any unsightly errors?

**Comment by transmissions11** [comment]
i think if it showed the invalid arg error it will probably show the unknown oid error too...

**Comment by onbjerg** [comment]
now i get `sysctl: cannot stat /proc/sys/sysctl/proc_translated: No such file or directory` :facepalm: let's redirect stderr

```
$(sysctl -n sysctl.proc_translated 2>/dev/null)
```

File: forgeup/install [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    *)
+      echo "forgeup: unsupported platform: $PLATFORM"
+      exit 1  
+      ;;
+  esac
+
+  ARCHITECTURE="$(uname -m)"
+
+  if [ "${ARCHITECTURE}" = "x86_64" ]; then
+    if [ "$(sysctl -n sysctl.proc_translated)" = "1" ]; then
```
Comment (onbjerg) [inline forgeup/install:47]:
```suggestion
    if [ "$(sysctl -n sysctl.proc_translated 2>/dev/null)" = "1" ]; then
```

**Comment by transmissions11** [comment]
kk @onbjerg try now?

**Comment by onbjerg** [comment]
works like a charm :1st_place_medal:

**Comment by transmissions11** [comment]
added docs

**Review by gakonst — APPROVED** [review]
epic

**Pull request merged**